Additional models

cconrad8 opened the pull request:
- added antibody, cell line, animal model, and genetic reagant + ui schemas

cconrad8 commented:
Tested some in here: https://sage-bionetworks.github.io/synapse-web-monorepo/?path=/story/ui-dynamicform--nf-data-submission-form

improvements can be made now or at a later time on labels such as 
![image](https://github.com/user-attachments/assets/f2086079-dd4d-484b-adc6-092ba550b254)

anngvu commented on NF-Tools-Schemas/antibody/submitAntibody.json:
```diff
@@ -0,0 +1,173 @@
+        ]
+      }
+    },
+    "alleleType": {
```
Also for alleleType

anngvu reviewed (changes requested):
Noted some suggested fixes.

cconrad8 commented:
Thanks @anngvu for your comments.  There are still some style issues I feel like I have - it's a long list of questions for genetic reagents and I know for the Gene Therapy initiative, Efren asked me how pepole will be able to easily submit multiple genetic reagents.

cconrad8 commented on NF-Tools-Schemas/genetic-reagant/submitGeneticReagant.json:
```diff
@@ -0,0 +1,145 @@
+  "title": "Submit Genetic Reagent",
+  "type": "object",
+  "properties": {
+    "geneticReagentName": {
```
added several more elements! Thanks for noticing

anngvu commented:
(Testing by viewing everything in rjsf playground)
Can confirm that for multiple reagents, it looks like it's set up to add another one 👍🏼 : 
 
<img width="300" alt="image" src="https://github.com/user-attachments/assets/4853c039-4140-40d1-bc36-0104ffa1952d">

<img width="300" alt="image" src="https://github.com/user-attachments/assets/049141d8-ed4a-47ad-bfcf-3285c9907d4c">

If I click on the '+', another card to fill in another genetic reagent is created. This was on the 'default' theme. the 'materialui 4' did look a little weird, but I think Jay's version of material UI should be fine.

anngvu commented on NF-Tools-Schemas/cell-line/submitCellLineUiSchema.json:
```diff
@@ -0,0 +1,88 @@
+      "ui:widget": "select",
+      "ui:placeholder": "Select a genetic mutation"
+    },
+    "species": {
```
Maybe add choices for human or mouse?

anngvu reviewed (changes requested):
Just a couple of things + https://github.com/nf-osi/nf-research-tools-schema/pull/70 (so I think everything can be merged today).

anngvu commented:
@cconrad8 Yeah, some things like genetic reagent _are_ still kind of long. But you're just defining the schema so that one day someone will connect their Benchling lab notebook which will have plasmid info already and you just transform the data.

allaway commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+    },
+    "backgroundSubstrain": {
+      "type": "string",
+      "title": "What is the background substrain?"
```
```suggestion
      "title": "What is the background substrain?"
```

Strains and substrains are a little confusing. Suggest adding an example here and for the strain question too

cconrad8 replied:
Added an example that pops up when you start to type - but maybe need to add a description below as well.

allaway commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+    },
+    "strainNomenclature": {
+      "type": "string",
+      "title": "What is the strain nomenclature?"
```
```suggestion
      "title": "If available, what is the official nomenclature of the strain (e.g. in mice, following the MGI standard nomenclature)?"
```

allaway commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+    },
+    "animalModelDisease": {
+      "type": "string",
+      "title": "What is the animal model disease?",
```
```suggestion
      "title": "What disease does this organism model?",
```

allaway commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+    },
+    "animalModelManifestation": {
+      "type": "string",
+      "title": "What is the animal model of manifestation?",
```
```suggestion
      "title": "What manifestations of the disease does this organism model?"
```

allaway commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+    },
+    "mutationTypes": {
+      "type": "array",
+      "title": "Do any of the mutation types apply?",
```
```suggestion
      "title": "Do any of the mutation types apply?",
```

suggest putting gene symbol question first

allaway commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+    },
+    "affectedGeneName": {
+      "type": "string",
+      "title": "What is the Affected Gene Name?"
```
```suggestion
      "title": "What is the Affected Gene Name?"
```

We can remove this - there are databases that have this information.

allaway commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+    },
+    "affectedGeneSymbol": {
+      "type": "string",
+      "title": "What is the Affected Gene Symbol?"
```
```suggestion
      "title": "What is the Affected Gene Symbol (e.g. in humans, NF1, or in mice, Nf1)?"
```

allaway commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+    },
+    "chromosomeNumber": {
+      "type": "string",
+      "title": "What chromosome number is the affected gene?"
```
```suggestion
      "title": "What chromosome is the affected gene on?"
```

anngvu replied:
Validated that this was removed

cconrad8 commented on NF-Tools-Schemas/animal-model/submitAnimalModel.json:
```diff
@@ -0,0 +1,190 @@
+        "Other"
+      ]
+    },
+    "mutationTypes": {
+      "type": "array",
+      "title": "Do any of the mutation types apply?",
+      "items": {
+        "type": "string",
+        "enum": [
+          "Disruption caused by insertion of vector",
+          "Duplication",
+          "Insertion",
+          "Insertion of gene trap vector",
+          "Intergenic deletion",
+          "Intragenic deletion",
+          "Inversion",
+          "Loss of heterozygosity (unspecified mechanism)",
+          "Nucleotide repeat expansion",
+          "Nucleotide substitutions",
+          "Single point mutation",
+          "Translocation",
+          "Transposon insertion",
+          "Viral insertion",
+          "Other"
+        ]
+      }
+    },
```
agreed!

cconrad8 commented:
Update: Converted questions to statements, removed antibody and cell line. (want to get a good working product first). Added if/then conditionality to vendor/investigator and developer.

anngvu reviewed (changes requested):
Conditional rendering looks great, and the questions and options make sense except for one section where I spent more time and thought was a little confusing (Bacterial Resistance).

cconrad8 commented on NF-Tools-Schemas/genetic-reagent/submitGeneticReagent.json:
```diff
@@ -0,0 +1,419 @@
+        "Ampicillin": {
+          "type": "boolean"
+        },
+        "AmpicillinandBleocin(Zeocin)": {
```
@anngvu good catch - should be fixed now!

anngvu commented:
Note: Some of the question re-wording no longer applied as Efren asked to not use the question format, I think. But some fields were removed to streamline collection.

allaway reviewed (approved):
We discussed in person hiding the "id" columns, e.g. geneticReagentId, because these are generated by us. Otherwise, LGTM! Thanks!
